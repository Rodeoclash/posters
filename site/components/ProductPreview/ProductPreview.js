import Link from "next/link";
import ProductAddToCart from "../ProductAddToCart";
import ProductImage from "../ProductImage";
import ProductInCart from "../ProductInCart";
import ProductLink from "../ProductLink";

import styles from "./ProductPreview.module.css";

const ProductPreview = ({ product }) => (
  <div className={styles.root}>
    <ProductLink product={product}>
      <ProductImage product={product} />
    </ProductLink>
    <div className={styles.meta}>
      <h3 className={styles.title}>{product.title}</h3>
      <ProductAddToCart product={product} />
      <ProductInCart product={product} />
    </div>
  </div>
);

export default ProductPreview;
