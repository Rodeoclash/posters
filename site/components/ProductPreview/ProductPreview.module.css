.root {
}

.meta {
  display: flex;
  margin: 0 -0.5rem;
}

.meta > * {
  margin: 0 0.5rem;
}

.title {
  flex-grow: 1;
}
