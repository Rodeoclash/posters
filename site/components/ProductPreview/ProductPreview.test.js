import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";

import ProductPreview from "./ProductPreview";

it("renders correctly", () => {
  const product = {
    title: "title",
  };

  const renderer = new ShallowRenderer();

  renderer.render(<ProductPreview product={product} />);

  const result = renderer.getRenderOutput();

  expect(result).toMatchSnapshot();
});
