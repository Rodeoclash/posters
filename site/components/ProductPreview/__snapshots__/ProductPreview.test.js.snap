// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <a
    href="/posters/test-poster-2"
    onClick={[Function]}
    onMouseEnter={[Function]}
  >
    <img
      src="https://cdn.shopify.com/s/files/1/0380/1713/8733/products/mockup-29dc6c6a.png?v=1585737094"
    />
  </a>
  <div>
    <h3>
      Test Poster 1
    </h3>
    <button
      onClick={[Function]}
    >
      Add to cart
    </button>
  </div>
</div>
`;
